src/main/java/it/pagopa/pn/national/registries/client/ipa/IpaClient.java

```diff
@@ -21,58 +18,30 @@
 import static it.pagopa.pn.national.registries.constant.ProcessStatus.PROCESS_SERVICE_WS23_PEC;
 
 @Component
+@RequiredArgsConstructor
 @lombok.CustomLog
 public class IpaClient {
 
-    private final WebClient webClient;
-
-    protected IpaClient(IpaWebClient ipaWebClient) {
-        webClient = ipaWebClient.init();
-    }
+    private final IpaApi ipaApi;
 
     public Mono<WS23ResponseDto> callEServiceWS23(String taxId, String authId) {
         log.logInvokingExternalDownstreamService(PnLogger.EXTERNAL_SERVICES.IPA, PROCESS_SERVICE_WS23_PEC);
-        return webClient.post()
-                .uri("/ws/WS23DOMDIGCFServices/api/WS23_DOM_DIG_CF")
-                .headers(httpHeaders -> httpHeaders.setContentType(MediaType.MULTIPART_FORM_DATA))
-                .body(BodyInserters.fromFormData(createRequestWS23(taxId, authId)))
-                .retrieve()
-                .bodyToMono(WS23ResponseDto.class)
+        return ipaApi.callEServiceWS23(taxId, authId)
                 .doOnError(throwable -> {
                     log.logInvokationResultDownstreamFailed(PnLogger.EXTERNAL_SERVICES.IPA, MaskDataUtils.maskInformation(throwable.getMessage()));
                     checkIPAException(throwable);
                 });
     }
 
-
-    private MultiValueMap<String, String> createRequestWS23(String taxId, String authId) {
-        LinkedMultiValueMap<String, String> requestWS23 = new LinkedMultiValueMap<>();
-        requestWS23.add("CF", taxId);
-        requestWS23.add("AUTH_ID", authId);
-        return requestWS23;
-    }
-
     public Mono<WS05ResponseDto> callEServiceWS05(String codAmm, String authId) {
         log.logInvokingExternalService(PnLogger.EXTERNAL_SERVICES.PN_NATIONAL_REGISTRIES, PROCESS_SERVICE_WS05_PEC);
-        return webClient.post()
-                .uri("/ws/WS05AMMServices/api/WS05_AMM")
-                .headers(httpHeaders -> httpHeaders.setContentType(MediaType.MULTIPART_FORM_DATA))
-                .body(BodyInserters.fromFormData(createRequestWS05(codAmm, authId)))
-                .retrieve()
-                .bodyToMono(WS05ResponseDto.class)
+        return ipaApi.callEServiceWS05(codAmm, authId)
                 .doOnError(throwable -> {
-                    log.logInvokationResultDownstreamFailed(PnLogger.EXTERNAL_SERVICES.INAD, MaskDataUtils.maskInformation(throwable.getMessage()));
+                    log.logInvokationResultDownstreamFailed(PnLogger.EXTERNAL_SERVICES.PN_NATIONAL_REGISTRIES, MaskDataUtils.maskInformation(throwable.getMessage()));
```

Comment:
corretto che l'external service sia PN_NATIONAL_REGISTRIES non possiamo essere piú precisi? IPA in questo caso

Reply:
Certo correggo subito.